include/bluetooth/conn.h

```diff
@@ -297,6 +297,29 @@ int bt_conn_security(struct bt_conn *conn, bt_security_t sec);
  */
 u8_t bt_conn_enc_key_size(struct bt_conn *conn);
 
+enum {
+	/* Authentication failed. */
+	BT_CONN_AUTH_ERR_AUTHENTICATION_FAIL,
```

Comment:
I would omit the term AUTHENTICATION from these since it is already part of the prefix BT_CONN_AUTH.

Reply:
The prefix is unfortunate in that case. This error code is to say that the authentication part of the pairing procedure failed. For example the user has compared the two different passkey and they did not match, so he pressed cancel. I feel like authentication failure is more precise than saying "fail"

Reply:
Not following you argument, the error domain is BT_CONN_AUTH_ERR and you say that is the authentication part thus I don't see why we need to keep authentication as error when the error domain is already refering to authentication part, actually even the conn part sounds a bit redundant here since authentication error can only happen on a connection.